kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -185,6 +186,12 @@ trait ProcBuilder {
     process
   }
 
+  def killApplicationWrap(engineRefId: String): String =
```

Comment:
I have an alternative idea for it, change the method signature to `def killApplication(clue: Either[String, String]): String`, WDYT @Nick-0723 @yaooqinn?

Reply:
So we judge whether it is Left or Right in EngineRef? We can't get logs in EngineRef

Reply:
Yeah, I mean merge two methods to one. And use `Left` to represent  `engineRefId` and Right to represent `line`

Reply:
the right will be removed eventually

Reply:
> Yeah, I mean merge two methods to one. And use `Left` to represent `engineRefId` and Right to represent `line`

If it is merged into one, the implementation of the flink engine will also needs to be modified

Reply:
I think we don't need such thing, the tag-based killing is promising

Reply:
@ulysses-you , I think this conversation will solve your question

Reply:
@ulysses-you , I think this conversation will solve your question

Reply:
I have read this history.. just think it's unnecessary